src/pages/admin/MiradorForm.js

```diff
@@ -0,0 +1,503 @@
+              enabled: false
+            }
+          };
+        } catch (error) {
+          console.error(error);
+        }
+        this.setState({
+          formState: siteInfo,
+          prevFormState: siteInfo,
+          site: site
+        });
+      } else {
+        let siteInfo = {
+          selectedTheme: "light",
+          language: "en",
+          id: "mirador_viewer",
+          window: {
+            allowClose: false,
+            allowFullscreen: true,
+            allowMaximize: false,
+            allowTopMenuButton: true,
+            allowWindowSideBar: true,
+            sideBarPanel: "info",
+            defaultView: "single",
+            sideBarOpen: false,
+            panels: {
+              info: true,
+              attribution: true,
+              canvas: false,
+              annotations: true,
+              search: false,
+              layers: false
+            }
+          },
+          windows: [
+            {
+              manifestId: ""
+            }
+          ],
+          thumbnailNavigation: {
+            defaultPosition: "far-bottom",
+            displaySettings: true,
+            height: 130,
+            width: 100
+          },
+          workspace: {
+            draggingEnabled: false,
+            allowNewWindows: false,
+            isWorkspaceAddVisible: false,
+            showZoomControls: true,
+            type: "mosaic"
+          },
+          workspaceControlPanel: {
+            enabled: false
+          }
+        };
+        this.setState({
+          formState: siteInfo,
+          prevFormState: siteInfo,
+          site: site
+        });
+      }
+    }
+  }
+
+  stringToBoolean = value => {
+    switch (value) {
+      case "true":
+        return true;
+      case "false":
+        return false;
+      default:
+        return value;
+    }
+  };
+
+  updateInputValue = event => {
+    let { name, value } = event.target;
+    value = this.stringToBoolean(value);
+    this.setState(prevState => {
+      return {
+        formState: { ...prevState.formState, [name]: value }
+      };
+    });
+  };
+
+  updateWindowValue = event => {
+    let { name, value } = event.target;
+    value = this.stringToBoolean(value);
+    if (parseInt(value)) {
+      value = parseInt(value);
+    }
+    this.setState(prevState => {
+      return {
+        formState: {
+          ...prevState.formState,
+          window: { ...prevState.formState.window, [name]: value }
+        }
+      };
+    });
+  };
+
+  updateThumbnailValue = event => {
+    let { name, value } = event.target;
+    value = this.stringToBoolean(value);
+    if (parseInt(value)) {
+      value = parseInt(value);
+    }
+    this.setState(prevState => {
+      return {
+        formState: {
+          ...prevState.formState,
+          thumbnailNavigation: {
+            ...prevState.formState.thumbnailNavigation,
+            [name]: value
+          }
+        }
+      };
+    });
+  };
```

Comment:
It could also merge and change. The difference between these are return {}

Reply:
I'm not sure I understand your question here. Could you elaborate, please?

Reply:
I realized you probably meant to combine these functions. I added a function to update both the window and thumbnailNavigation objects since they are both at the same level in the state object. The updateInput function is harder to combine because it is at a different level and I don't have a solution for how to handle that. However, so far all the config components I've written have this same redundancy so it is actually consistent with the rest of the code in the app. Perhaps this PR could be merged and then on a rainy day in the future someone can go back and streamline my clunky code in all the config components?